### PR TITLE
chore!: remove replacing values in configuration marked by 'replace' suffix

### PR DESCRIPTION
**Pull request by kkujawa-sumo**
chore: remove replacing values in configuration marked by 'replace' suffix
make values in configuration for otelcol dependended on otelcol parameters in values.yaml

removes "otelcol.k8s.cluster.name"  which was added in #1721 and it is no longer in use

this pull request reverts changes introduced in #1721

related to #1729

Generated templates have been compared using output of `helm template . -f values.yaml -f values_custom.yaml ` where 
`values_custom.yaml` had following form:

```yaml
sumologic:
  ## Sumo access ID
  accessId: "dummy"

  ## Sumo access key
  accessKey: "dummy"

  ### Traces configuration
  ## Set the enabled flag to true to enable traces ingestion.
  traces:
    enabled: true

```
related change in `update-collection-v3`:  https://github.com/SumoLogic/sumologic-kubernetes-tools/pull/382

### CODE REVIEW

**Review by sumo-drosiek — APPROVED** [review]
Great. Not sure if we need a migration for that change.

**Comment by kkujawa-sumo** [comment]
> Great. Not sure if we need a migration for that change.

I need to add at least information to migration doc that the mechanism with replacing some configuration values with 'replace' suffix was removed.

File: docs/v3-migration-doc.md [hunk 1]
```diff
@@ -18,6 +19,24 @@ In this document we detail the changes as well as the exact steps for migration.
...
+  - `processors.source.category_prefix.replace`
+  - `processors.source.category_replace_dash.replace`
+  - `processors.source.exclude_namespace_regex.replace`
+  - `processors.source.exclude_pod_regex.replace`
+  - `processors.source.exclude_container_regex.replace`
+  - `processors.source.exclude_host_regex.replace`
+  - `processors.resource.cluster.replace`
+  - `exporters.sumologic.source_name.replace`
+  - `exporters.sumologic.source_category.replace`
+
```
[the remark this answers is not in the document]
Comment (kkujawa-sumo) [reply docs/v3-migration-doc.md]: I shortened the introduction ;)

**Review by sumo-drosiek — APPROVED** [review]
:tada:

**Pull request merged**